camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/ConfigInheritanceYamlTest.java

```diff
@@ -0,0 +1,316 @@
+                "location: localhost-stub",
+                "services:",
+                "- type: EmptySoftwareProcess-with-conf");
+        
+        Entity app = createStartWaitAndLogApplication(new StringReader(yaml));
+        Entity entity = Iterables.getOnlyElement(app.getChildren());
+        
+        assertEmptySoftwareProcessConfig(
+                entity,
+                ImmutableMap.of("ENV1", "myEnv1"),
+                ImmutableMap.of(emptyFile.toUri().toString(), "myfile"),
+                ImmutableMap.of("mykey", "myval", "templateOptions", ImmutableMap.of("myOptionsKey", "myOptionsVal")));
+    }
+    
+    @Test
+    public void testInheritsParentConfig() throws Exception {
+        String yaml = Joiner.on("\n").join(
+                "location: localhost-stub",
+                "services:",
+                "- type: org.apache.brooklyn.entity.stock.BasicApplication",
+                "  brooklyn.config:",
+                "    shell.env:",
+                "      ENV1: myEnv1",
+                "    templates.preinstall:",
+                "      "+emptyFile.toUri()+": myfile",
+                "    files.preinstall:",
+                "      "+emptyFile.toUri()+": myfile",
+                "    templates.install:",
+                "      "+emptyFile.toUri()+": myfile",
+                "    files.install:",
+                "      "+emptyFile.toUri()+": myfile",
+                "    templates.runtime:",
+                "      "+emptyFile.toUri()+": myfile",
+                "    files.runtime:",
+                "      "+emptyFile.toUri()+": myfile",
+                "    provisioning.properties:",
+                "      mykey: myval",
+                "      templateOptions:",
+                "        myOptionsKey: myOptionsVal", 
+                "  brooklyn.children:",
+                "  - type: org.apache.brooklyn.entity.software.base.EmptySoftwareProcess");
+
+        Entity app = createStartWaitAndLogApplication(new StringReader(yaml));
+        Entity entity = Iterables.getOnlyElement(app.getChildren());
+        
+        assertEmptySoftwareProcessConfig(
+                entity,
+                ImmutableMap.of("ENV1", "myEnv1"),
+                ImmutableMap.of(emptyFile.toUri().toString(), "myfile"),
+                ImmutableMap.of("mykey", "myval", "templateOptions", ImmutableMap.of("myOptionsKey", "myOptionsVal")));
+    }
+    
+    /**
+     * TODO This hangs because the attributeWhenReady self-reference is resolved against the entity
+     * looking up the config value (i.e. the child). Therefore it waits for the TestEntity to have
+     * a value for that sensor, but this never happens. The way to avoid this is to explicitly set
+     * the component that the attributeWhenReady should apply to (e.g. see {@link #testInheritsParentConfigTask()}.
+     * 
+     * Do we want to just exclude this test? Or do we want to "fix" it? Which entity should the
+     * attributeWhenReady apply to?
```

Comment:
+1 for fixing it. We have a similar problem with resource references where they will be resolved against the libraries of the catalog item using the resource, not the one that's declaring it. Need to tag config with the place where it was declared.